api/app/controllers/spree/api/customer_returns_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Spree
+  module Api
+    class CustomerReturnsController < Spree::Api::BaseController
+      before_action :load_order
+      around_action :lock_order, only: [:create, :update, :destroy, :cancel]
+
+      rescue_from Spree::Order::InsufficientStock, with: :insufficient_stock_error
+
+      def create
+        authorize! :create, CustomerReturn
+        @customer_return = CustomerReturn.create(customer_return_params)
+        if @customer_return.save
+          respond_with(@customer_return, status: 201, default_template: :show)
+        else
+          invalid_resource!(@customer_return)
+        end
+      end
+
+      def index
+        authorize! :admin, CustomerReturn
```

Comment:
I think all actions should require admin authorization, so maybe can use `authorize_resource` on top for set the authorization for all actions

Reply:
We had to deal with a lot of permission sets in solidus marketplace extension, `:admin` is not that friendly when it comes to grant specific actions, maybe use the same idea than resources controller and authorize each action by its name? that way we can allow to show customer returns to specific roles but without the ability to update the record

Reply:
@softr8 Yes good point, I'll make that change and push shortly! (also have to update the documentation which will take some time 😄 ) 

@ccarruitero Sorry for taking so long to get back to you! I think `authorize_resource` _could_ be something that simplifies a lot of our API actions, but we don't use it anywhere in our code currently, and I'd prefer to stick with existing conventions for this PR. However, if you want to open an issue to migrate to using `authorize_resource`, please do! I think it would be a welcome addition, removing lots of lines of code from the controllers that currently authorize on a per-action basis.